### PR TITLE
3143 fetching patient from central

### PR DESCRIPTION
**Pull request by clemens-msupply**
Part of #3143 

# 👩🏻‍💻 What does this PR do? 

- Refactor/create a hook to fetch a patient. I found the previous implementation a bit messy and hope this makes it clearer.
- Enable LIKE patient search for central server patients

There are still some issues I like to look at, at a later stage because I can't figure them out right now:
- After moving the patient over to the local site. The search endpoint fails to return it even though it is in the database and a full page reload fixes it!
- I can't get tanquery to refetch the main patient list automatically even though I invalidated the query key...

### CODE REVIEW

**Comment by mark-prins** [comment]
I'm not seeing the same results as you - the call to link the patient is made and then a call is made to fetch the patient list again. 

However - the sync doesn't work, as only the `name_store_join` record is added to `sync_out`
This gives
```
2024-03-09 12:58:25.677451000 [WARN] <service::sync::translation_and_integration:107>:Failed to get name '582fdcfd-cc60-40b1-be52-ecf93a43ba03' for name_store_join '46B9FBFD8C004F01B7FE987609FA68F9' "46B9FBFD8C004F01B7FE987609FA68F9" "name_store_join"
```

Perhaps my configuration is incorrect? I've followed the steps in the issue, but that doesn't say too much about how to configure.

**Comment by clemens-msupply** [comment]
Yeah I fear there are still a couple of problems:
- there seem to be a regression in central: https://github.com/msupply-foundation/open-msupply/issues/3143
- I can't invalidate tanquery correctly (this PR should have fixed it, but the main patient list is still not refreshing)
- There was another API problem searching for the new patient after fetching it from central (but maybe that is related to the first point)

Probably best to wait till central is fixed?

**Comment by clemens-msupply** [comment]
Central is working and I fixed polling for the sync status. I.e. invalidateQueries is now called after the patient has been synced!

**Review by mark-prins — APPROVED** [review]
Thanks @clemens-msupply 👍

File: client/packages/system/src/Patient/CreatePatientModal/PatientResultsTab.tsx [hunk 1]
```diff
@@ -181,6 +181,12 @@ export const PatientResultsTab: FC<PatientPanel & { active: boolean }> = ({
     setData(patients);
   }, [localSearchData, centralSearchData]);
 
+  const onClose = useCallback(() => {
```
[the remark this answers is not in the document]
Comment (clemens-msupply) [reply client/packages/system/src/Patient/CreatePatientModal/PatientResultsTab.tsx:184]: Think I was looking into some (potential) re-rendering issue and tried to keep this stable. Not sure if it really is needed. Should I investigate?

[the remark this answers is not in the document]
Comment (mark-prins) [reply client/packages/system/src/Patient/CreatePatientModal/PatientResultsTab.tsx:184]: I doubt that it adds any value, I also doubt that it causes a problem - so I'd leave it as is. The change was working after all!

**Comment by regotaina** [comment]
Tested on 1.7RC4

Passed the test 🥇 💯

**Pull request merged**